scripts/build.py

```diff
@@ -0,0 +1,213 @@
+
+"""
+
+import subprocess
+import os
+import sys
+import multiprocessing
+import argparse
+
+
+def run_command(cmd, cwd=None, as_root_user=False, collect_output=False):
+    """Run a command
+
+    Args:
+        cmd (:obj:`list` of :obj:`str`): Command to run as a list object
+                                         Eg: ['ls', '-l']
+        cwd (str): Directory to execute the command from
+        as_root_user (bool): Run command as root user with admin priveleges (supported on mac and linux)
+        collect_output (bool): Get output from the command. The string returned from this function
+                               call will have the output from command execution.
+    Raises:
+        (IOError): If command errored out
+    Returns:
+        (str|None): If collect_output is provided, a string is returned
+                    None if output is not requested
+    """
+    if as_root_user and (is_mac_os() or is_linux_os()):
+        cmd.insert(0, 'sudo')
+
+    cmd_string = ' '.join(cmd)
+    print('Running cmd: {0}\n'.format(cmd_string))
+    try:
+        if collect_output:
+            output = subprocess.check_output(cmd, cwd=cwd)
+            return output.decode('utf-8').strip()
+        else:
+            subprocess.call(cmd, cwd=cwd)
+    except subprocess.CalledProcessError:
+        raise IOError('Error executing {0}'.format(cmd_string))
+
+
+def get_repo_root_dir():
+    """ Get root directory of current git repo
+
+    This function takes the guesswork out when executing commands
+    or locating files relative to root of the repo
+
+    Raises:
+        IOError: Current directory is not in a git repo
+
+    Returns:
+        (str): Root directory of git repo
+    """
+    return run_command(['git', 'rev-parse', '--show-toplevel'], collect_output=True)
+
+
+REPO_ROOT_DIR = get_repo_root_dir()
+
+def is_windows_os():
+    """Check if the current OS is Windows
+
+    Returns:
+        (bool): True if windows, False otherwise
+    """
+    return sys.platform.startswith('win')
+
+
+def is_linux_os():
+    """Check if the current OS is Linux
+
+    Returns:
+        (bool): True if windows, False otherwise
+    """
+    return sys.platform.startswith('linux')
+
+
+def is_mac_os():
+    """Check if the current OS is mac
+
+    Returns:
+        (bool): True if windows, False otherwise
+    """
+    return sys.platform.startswith('darwin')
+
+
+def get_vcpkg_triplet(arch):
+    """ Get vcpkg target triplet (platform definition)
+
+    Args:
+        arch (str): Architecture (eg: 'x86', 'x64')
+
+    Raises:
+        ValueError: If current OS is not win,mac or linux
+    
+    Returns:
+        (str): Triplet name
+               Eg: "x64-windows-static"
+    """
+    triplet_name = [arch]
+    if is_windows_os():
+        triplet_name.append('windows')
+        # For windows, default is to build dynamic. Hence we specify static.
+        # For mac/linux, default is to build static libraries
+        triplet_name.append('static')
```

Comment:
Could we output these decisions to the console so that any automatic configurations are evident in the build results?